### DISCUSSION TITLE
Unable to open the Airflow UI after updating

### DISCUSSION THREAD

**Discussion by rsotogar** [apache/airflow#19394] [Q&A]
After updating the requirements file and updating the Airflow environment in MWAA the UI become unresponsive.

Airflow version: 2.0.2
Requirements file:

gspread==3.7.0
google-api-python-client==1.12.8
apache-airflow-providers-amazon==1.4.0
apache-airflow-providers-apache-spark==1.0.3
apache-airflow-providers-jdbc==1.0.1
apache-airflow-providers-mysql==1.1.0
apache-airflow-providers-postgres==1.0.2
apache-airflow-providers-slack==3.0.0
apache-airflow-providers-google==6.0.0
apache-airflow-providers-salesforce==3.2.0
s3fs<=0.4
pyarrow==4.0.1
google-trans-new==1.1.9
nltk==3.6.2
scikit-learn==0.24.2
facebook_business==11.0.0
facebook-sdk==3.1.0
google-ads==12.0.0

![Uploading Screenshot 2021-11-03 at 18.24.59.png…]()

**Comment by potiuk** [comment]
I am afraid you should check with MWAA support

**Comment by potiuk** [comment]
I think the reason is likely the same as #19391  - Google providers (and other providers in your list) do not support Airflow 2.0.2  - you need 2.1.0+ for them. This likely causes compatibility issues and pip conflicts when you try to install them (installing provider that require 2.1.0 will attempt to install newer version of airflow which needs to run migration etc. (and for MWAA you likely need to have full support for newer version of Airflow to be able to upgrade  to that version, and likely it should be done using tools that MWAA provides).

**Reply by potiuk** [reply]
But you should check with MWAA support, I think as I am not aware of the internals they have

**Reply by john-jac** [reply]
As with all Airflow installations [MWAA recommends ](https://docs.aws.amazon.com/mwaa/latest/userguide/working-dags-dependencies.html#working-dags-dependencies-test-create)using the [constraint file](https://airflow.apache.org/docs/apache-airflow/2.0.2/installation.html#constraints-files) at the top of requirements to ensure compatibility.  When I [test the above requirements](https://github.com/aws/aws-mwaa-local-runner) against that guidance I see:
```
ERROR: Cannot install apache-airflow-providers-amazon==1.4.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested apache-airflow-providers-amazon==1.4.0
    The user requested (constraint) apache-airflow-providers-amazon==1.3.0
```

**Comment by rsotogar** [comment]
I’m not sure since I rolled the changes I made back to the previous requirements.txt file and still doesn’t work.

The problem arose when I added the Google dependency and updated the MWAA environment. I expected that when changes are rolled back the airflow ui would open again, but it didn’t.

Dr. Ramon Soto

**Reply by john-jac** [reply]
@potiuk is it possible that if a library has a dependency on a 2.1 version of Airflow that it could somehow upgrade the DB and make it unusable by 2.0?

**Reply by potiuk** [reply]
No, that should not happen unless somewhere in the "update environment" process, `airflow db upgrade` is explicitly called externally by the "update environment process". 

Since the new providers have "apache-airflow >=2.1" dependency, if you install them, they will automatically install latest version of airlfow from PyPI (unless you make some protection). It's besicaly the same as if you add `apache-airlflow==2.2.1` to your requirements (but transitively). But airflow does not do any migrations on its own. 

Current  behaviour is that airflow components will randomly fail if you have non-migrated database, but it will never upgrade the db unless 'airflow db upgrade` command is run manually.

We toyed with this idea here to run migrations automatically (not yet merged): https://github.com/apache/airflow/pull/18439, but (we correctly) agreed between a number of commiters that this would be too dangerous (and this case proves it would be). The final shape of the PR is that when you run aiflow command with a terminal (so during development) we will ask question - with timeout - whether to migrate, and fail if the user does not explicitly answer y. But for unnattended execution of airlfow commands we will always fail immediately if the migration is not done. This approach will be much better diagnostic-wise (no random error, explicit "please migrate" message) but this is not even merged yet.

**Comment by rsotogar** [comment]
So I ended up making a copy of my current environment since setting the contraints file did not help.

I don't know why this happened, but in the future I will be more careful about Airflow dependencies & versions.

Thanks!
